packages/react-menu/src/components/MenuTrigger/useTriggerElement.ts

```diff
@@ -92,21 +92,29 @@ export const useTriggerElement = (state: UseTriggerElementState): MenuTriggerSta
   });
 
   const disabled = child.props?.disabled;
-  const triggerProps: Partial<React.HTMLAttributes<HTMLElement>> = {
+  const triggerProps: MenuTriggerChildProps = {
     'aria-haspopup': true,
     'aria-expanded': open,
     id: triggerId,
-    ...(child.props || {}),
-
-    // These handlers should always handle the child's props
-    ...(!disabled && {
-      // These handlers should always handle the child's original handlers
-      onClick,
-      onMouseEnter,
-      onContextMenu,
-      onKeyDown,
-      onBlur,
-    }),
+    // spread props here because below event handlers must handle original prop
+    ...child.props,
+
+    ...(!disabled
+      ? {
+          onClick,
+          onMouseEnter,
+          onContextMenu,
+          onKeyDown,
+          onBlur,
+        }
+      : // Spread disabled event handlers to implement contract and avoid specific disabled logic in handlers
+        {
+          onClick: () => null,
```

Comment:
can we create a `noop` function on the file scope to have the callback referentially stable?

Reply:
yeah that's a good idea, I overlooked it